app/views/posts/index.html.erb

```diff
@@ -0,0 +1,27 @@
+<p id="notice"><%= notice %></p>
+
+<div class="container d-flex flex-column justify-content-center w-100 bg-light py-2">
+
+  <h1 class="text-center">Posts</h1>
+    <% @posts.each do |post| %>
+    <div class="text-center d-flex justify-content-center my-3">
+      <div class="card col-6">
+        <div class="card-body">
+          <h1><%= link_to post.title, post_path(post) %></h1>
+          <% if user_signed_in? %>
+            <p>By: <%= post.user.username %> </p>
+          <% else %>
+            <p>By: Anonymous </p>
+          <% end %>
+          <p><%= truncate(post.body, length: 50) %></p>
+        </div>
+        <div>    
+          <% if user_signed_in? %>    
+            <%= link_to 'Edit', edit_post_path(post), class: 'btn btn-warning text-white mb-2' %>
+            <%= link_to 'Destroy', post, method: :delete, data: { confirm: 'Are you sure?' } , class: 'btn btn-danger text-white mb-2'%>
+          <% end %>
```

Comment:
View-related logic can be encapsulated in View helpers (app/helpers/\*_helper.rb)